### PR TITLE
Add chmod to auto perm script

### PR DESCRIPTION
**Pull request by stavros-k**
**Description**

⚒️ Fixes  # 

Renamed `permissions` to `ownership`
And added chmod (mentioning `permissions`)

Did some formatting, mostly for syntax highlighting to pop up

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [x] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [x] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

### CODE REVIEW

**Review by Ornias1993 — CHANGES_REQUESTED** [review]
- We should utilise the nfs4xdr_winacl equivalent for chmod on SCALE.
- Could you restyle the responses/warnings in console a tad? I think the iX log output is a tad cleaner...

File: library/common/templates/lib/util/_autoperms.tpl [hunk 1]
```diff
@@ -108,13 +112,15 @@ spec:
             {{- range $name, $hpm := $hostPathMounts }}
               {{- if $hpm.chmod }}
               echo "Automatically correcting permissions for {{ $hpm.mountPath }}..."
-              chmod ${{ $hpm.chmod }} /mounts/{{ $name }} || echo "Failed setting permissions using chmod..."
-              {{- end }}
+              chmod {{ $hpm.chmod }} /mounts/{{ $name }} || echo "Failed setting permissions using chmod..."
```
Comment (Ornias1993) [inline library/common/templates/lib/util/_autoperms.tpl:115]: The own seems to be recursive while the chmod does not...
Maybe we want recursive as an option? 

though that might mean we want to indent those three settings under a dict like this:
```
autoPermissions:
  chown: true
  chmod: 775
  recursive: true
  ```

Comment (stavros-k) [reply library/common/templates/lib/util/_autoperms.tpl:115]: Actually there are more things to do here...
We can only apply chmod/chown in hostPath, due to how volumes are rendered .in the container

While chown is not needed in pvc, chmod does (in rare cases, see Nextcloud).

Comment (stavros-k) [reply library/common/templates/lib/util/_autoperms.tpl:115]: > The own seems to be recursive while the chmod does not... Maybe we want recursive as an option?
> 
> though that might mean we want to indent those three settings under a dict like this:
> 
> ```
> autoPermissions:
>   chown: true
>   chmod: 775
>   recursive: true
> ```

But I'm cool with the dict

Comment (Ornias1993) [reply library/common/templates/lib/util/_autoperms.tpl:115]: > Actually there are more things to do here... We can only apply chmod/chown in hostPath, due to how volumes are rendered .in the container
> 
> While chown is not needed in pvc, chmod does (in rare cases, see Nextcloud).

Fair pont there indeed!
Maybe split the logic?

Comment (Ornias1993) [reply library/common/templates/lib/util/_autoperms.tpl:115]: Agreed :)

**Comment by stavros-k** [comment]
> * We should utilise the nfs4xdr_winacl equivalent for chmod on SCALE.
> * Could you restyle the responses/warnings in console a tad? I think the iX log output is a tad cleaner...

What are the equivalent of chmod in winacl?

Will do the logging too

**Comment by Ornias1993** [comment]
> > * We should utilise the nfs4xdr_winacl equivalent for chmod on SCALE.
> > * Could you restyle the responses/warnings in console a tad? I think the iX log output is a tad cleaner...
> 
> What are the equivalent of chmod in winacl?
> 
> Will do the logging too

You can ask caleb if you like :)

**Review by Ornias1993 — APPROVED** [review]

**Pull request merged**